### PR TITLE
Allow detaching loose building items

### PR DESCRIPTION
**Pull request by Bumber64**
Allows detaching workshop clutter, etc., so it can be autodumped.

### CODE REVIEW

**Review by myk002 — COMMENTED** [review]
looks good in theory. Have you identified places in our code that we can simplify with these functions?

**Comment by Bumber64** [comment]
> looks good in theory. Have you identified places in our code that we can simplify with these functions?

``detach_item`` already existed, so this just causes it not to fail for items in buildings. Looks like ``utils.getBuildingName`` does what the other function would've done, so I've reverted that.

**Comment by myk002** [comment]
one place I'm aware of is `build-now`. It has custom logic for detaching clutter. That could be removed, I think.

File: docs/changelog.txt [hunk 1]
```diff
@@ -90,6 +90,7 @@ Template for new versions:
 - ``Gui::makeAnnouncement``, ``Gui::autoDFAnnouncement``: don't display popup for all announcement types
 - ``gui.View:getMouseFramePos``: function now detects the correct coordinates even when the widget is nested within other frames
 - `strangemood`: correctly recognize Stonecutter and Stone Carver as moodable skills, move the Mason's boosted mood chance to the Stone Carver, and select Fell/Macabre based on long-term stress
+- Items module fn ``detach_item`` can now detach items loosely stored in buildings. Impacts ``autodump``, ``Items::moveToGround``, etc.
```
Comment (myk002) [inline docs/changelog.txt:93]: needs to get moved up to "Future"

**Review by myk002 — COMMENTED** [review]
could you remove the now redundant code in `build-now` to take advantage of this?

**Review by myk002 — APPROVED** [review]

**Pull request merged**